src/main/java/com/gradle/develocity/bamboo/GradleBuildScanInjector.java

```diff
@@ -123,13 +128,30 @@ private void inject(BuildContext buildContext, Collection<RuntimeTaskDefinition>
         LOGGER.debug("Develocity Gradle auto-injection completed");
     }
 
-    private void prepareEnvironment(BuildContext buildContext, GradleConfiguration config) {
+    private void prepareEnvironment(BuildContext buildContext, GradleConfiguration config, Collection<RuntimeTaskDefinition> tasks) {
         VariableContext variableContext = buildContext.getVariableContext();
 
-        Objects.runIfNotNull(config.server, s -> variableContext.addLocalVariable("DEVELOCITY_PLUGIN_DEVELOCITY_URL", s));
+        Objects.runIfNotNull(config.server, it -> variableContext.addLocalVariable("DEVELOCITY_PLUGIN_DEVELOCITY_URL", it));
         Objects.runIfTrue(config.allowUntrustedServer, () -> variableContext.addLocalVariable("DEVELOCITY_PLUGIN_DEVELOCITY_ALLOW_UNTRUSTED_SERVER", "true"));
-        Objects.runIfNotNull(config.develocityPluginVersion, v -> variableContext.addLocalVariable("DEVELOCITY_PLUGIN_DEVELOCITY_PLUGIN_VERSION", v));
-        Objects.runIfNotNull(config.ccudPluginVersion, v -> variableContext.addLocalVariable("DEVELOCITY_PLUGIN_CCUD_PLUGIN_VERSION", v));
-        Objects.runIfNotNull(config.pluginRepository, r -> variableContext.addLocalVariable("DEVELOCITY_PLUGIN_GRADLE_PLUGIN_REPOSITORY_URL", r));
+        Objects.runIfNotNull(config.develocityPluginVersion, it -> variableContext.addLocalVariable("DEVELOCITY_PLUGIN_DEVELOCITY_PLUGIN_VERSION", it));
+        Objects.runIfNotNull(config.ccudPluginVersion, it -> variableContext.addLocalVariable("DEVELOCITY_PLUGIN_CCUD_PLUGIN_VERSION", it));
+        Objects.runIfNotNull(config.pluginRepository, it -> variableContext.addLocalVariable("DEVELOCITY_PLUGIN_GRADLE_PLUGIN_REPOSITORY_URL", it));
+
+        Objects.runIfNotNull(
+                config.pluginRepositoryCredentialName,
+                it -> {
+                    UsernameAndPassword credentials = credentialsProvider.findByName(it).orElse(null);
+                    if (credentials == null) {
+                        LOGGER.warn("Plugin repository credentials with the name {} are not found.", it);
+                    } else {
+                        if (credentials.getUsername() == null || credentials.getPassword() == null) {
+                            LOGGER.warn("Plugin repository credentials {} do not have username or password set.", it);
+                        } else {
+                            variableContext.addLocalVariable("DEVELOCITY_PLUGIN_GRADLE_PLUGIN_REPOSITORY_USERNAME", credentials.getUsername());
+                            variableContext.addLocalVariable("DEVELOCITY_PLUGIN_GRADLE_PLUGIN_REPOSITORY_PASSWORD", credentials.getPassword());
```

Comment:
Do we know if this `VariableContext` is shown somewhere in the UI or logs? If so, will the password be masked?

Reply:
Hm, I'll see if I can find them somewhere.

Reply:
Didn't find it anywhere in the UI or logs of the job (our init script logic doesn't log it)